src/services/acls.ts

```diff
@@ -83,47 +84,51 @@ export const usePermissionsService = (
     return enhanceAclBindingListPage(response);
   };
 
-  const addPermission = async (acl: AclBinding) => {
+  const addPermission = useCallback(
+    async (acl: AclBinding) => {
+      const accessToken = await config?.getToken();
+      const api = new AclsApi(
+        new Configuration({
+          accessToken,
+          basePath: config?.basePath,
+        })
+      );
+      await api.createAcl(acl);
+    },
+    [config]
+  );
+
+  const deletePermission = useCallback(
+    async (acl: AclFilter) => {
+      const accessToken = await config?.getToken();
+      const api = new AclsApi(
+        new Configuration({
+          accessToken,
+          basePath: config?.basePath,
+        })
+      );
+      await api.deleteAcls(
+        acl.resourceType,
+        acl.resourceName,
+        acl.patternType,
+        acl.principal,
+        acl.operation,
+        acl.permissionType
+      );
+    },
+    [config]
+  );
+
+  const getResourceOperations = useCallback(async () => {
     const accessToken = await config?.getToken();
     const api = new AclsApi(
       new Configuration({
         accessToken,
         basePath: config?.basePath,
       })
     );
-    await api.createAcl(acl);
-  };
-
-  const deletePermission = async (acl: AclFilter) => {
-    const accessToken = await config?.getToken();
-    const api = new AclsApi(
-      new Configuration({
-        accessToken,
-        basePath: config?.basePath,
-      })
-    );
-    await api.deleteAcls(
-      acl.resourceType,
-      acl.resourceName,
-      acl.patternType,
-      acl.principal,
-      acl.operation,
-      acl.permissionType
-    );
-  };
-
-  const getResourceOperations = async () => {
-    const accessToken = await config?.getToken();
-    const api = new AclsApi(
-      new Configuration({
-        accessToken,
-        basePath: config?.basePath,
-      })
-    );
-    return await api
-      .getAclResourceOperations()
-      .then((response) => response.data);
-  };
+    return api.getAclResourceOperations().then((response) => response.data);
+  }, [config]);
 
   return {
```

Comment:
wrapping this return in [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo) should protect us from other places where the full object is being used as a dependency for an hook